### PR TITLE
Add DCOS to getting started guide list (docs)

### PR DESCRIPTION
**Pull request by karlkfi**
forgot to add the dcos guide to the list when it got added.

### CODE REVIEW

**Comment by karlkfi** [comment]
@davidopp Any idea who should review this?

**Comment by karlkfi** [comment]
Rebased, and filled in some of the empty columns.

File: docs/getting-started-guides/README.md [hunk 1]
```diff
@@ -147,6 +153,7 @@ libvirt              | custom       | Fedora | flannel     | [docs](fedora/flann
 KVM                  | custom       | Fedora | flannel     | [docs](fedora/flannel_multi_node_cluster.md)      |          | Community ([@aveshagarwal](https://github.com/aveshagarwal))
 Mesos/Docker         | custom       | Ubuntu | Docker      | [docs](mesos-docker.md)                           | [✓][4]  | Community ([Kubernetes-Mesos Authors](https://github.com/mesosphere/kubernetes-mesos/blob/master/AUTHORS.md))
 Mesos/GCE            |              |        |             | [docs](mesos.md)                                  |          | Community ([Kubernetes-Mesos Authors](https://github.com/mesosphere/kubernetes-mesos/blob/master/AUTHORS.md))
```
Comment (davidopp) [inline docs/getting-started-guides/README.md]: I realize it's not part of this PR, but I assume the OS should be Debian and the Networking should be custom or Docker? What sould config mgmt. be?

**Comment by davidopp** [comment]
(other than that, looks fine)

File: docs/getting-started-guides/README.md (hunk 1 above)
Comment (karlkfi) [inline docs/getting-started-guides/README.md]: The mesos guide doesn't come with automated cluster deployment scripts. It describes a manual process. So there's no config management, operating system, or networking specified. They could all be N/A, but we left them blank instead.

**Comment by karlkfi** [comment]
rebased to pick up unit test fix

**Comment by karlkfi** [comment]
Adding lgtm based on @davidopp's review.

**Comment by karlkfi** [comment]
Cherry-pick to release-1.1: #18034
